src/Automation/SnapshotCommand.cs

```diff
@@ -88,29 +73,11 @@ public static SnapshotCommandResult Execute(Config config, IOutputFileHelper out
                                 if (locationHelper.IsSarifExtension())
                                     // SaveAction.SaveSarifFile(outputFileHelper.GetNewSarifFilePath(), ec2.Id, !locationHelper.IsAllOption());
 #endif
-                                }
-
-                                string summaryMessage;
 
-                                if (accumulator.MayHaveErrors)
-                                {
-                                    summaryMessage = string.Format(CultureInfo.InvariantCulture, DisplayStrings.SnapshotDetailViolationsFormat, accumulator.Failed, accumulator.Inconclusive, a11yTestOutputFile);
-                                }
-                                else
-                                {
-                                    summaryMessage = DisplayStrings.SnapshotDetailNoViolationsDataDiscarded;
+                                    results.OutputFile = (a11yTestOutputFile, null);
```

Comment:
>(a11yTestOutputFile, null) [](start = 57, length = 26)

This line is pre-set for when we add the Sarif 2.1 output, but it will require touching if/when a third output file is added (Sarif 3.0, maybe). It may be overengineering, but I'd love to see all of the OutputFile creation be pulled into a tiny factory class, so that if/when new formats are added, the only change in inside that factory class. This like would be something like:

results.OutputFile = OutputFileFactory.FromA11yTestOutputFile(a11yTestOutputFile);

Reply:
I can definitely get behind the idea. If we don't have time to make such a change during this feature, we can do it when we add the sarif format.

Reply:
I can definitely get behind this idea. If we don't have time to make the change during this feature. We should do that when we add sarif.